lib/philomena/filters/elasticsearch_index.ex

```diff
@@ -0,0 +1,93 @@
+defmodule Philomena.Filters.ElasticsearchIndex do
+  @behaviour Philomena.ElasticsearchIndex
+
+  @impl true
+  def index_name do
+    "filters"
+  end
+
+  @impl true
+  def mapping do
+    %{
+      settings: %{
+        index: %{
+          number_of_shards: 5,
+          max_result_window: 10_000_000
+        }
+      },
+      mappings: %{
+        dynamic: false,
+        properties: %{
+          id: %{type: "integer"},
+          created_at: %{type: "date"},
+          user_id: %{type: "keyword"},
+          creator: %{type: "keyword"},
+          # boolean
+          public: %{type: "keyword"},
+          # boolean
+          system: %{type: "keyword"},
+          name: %{
+            type: "text",
+            analyzer: "snowball",
+            fields: %{
+              raw: %{
+                type: "keyword"
+              }
+            }
+          },
+          description: %{type: "text", analyzer: "snowball"},
+          spoilered_count: %{type: "integer"},
+          hidden_count: %{type: "integer"},
+          spoilered_tag_ids: %{type: "keyword"},
+          hidden_tag_ids: %{type: "keyword"},
+          spoilers_namespaced_tags: %{
+            properties: %{
+              name: %{type: "keyword"},
+              name_in_namespace: %{type: "keyword"},
+              namespace: %{type: "keyword"}
+            }
+          },
+          hides_namespaced_tags: %{
+            properties: %{
+              name: %{type: "keyword"},
+              name_in_namespace: %{type: "keyword"},
+              namespace: %{type: "keyword"}
+            }
+          },
+          hidden_complex_str: %{type: "wildcard"},
```

Comment:
The `wildcard` field type belongs to X-Pack. X-Pack is available for free but does not contain a compatible license with the project. This should use `keyword` instead.